src/rdo.rs

```diff
@@ -209,6 +227,68 @@ fn compute_rd_cost(
   (distortion as f64) + lambda * rate
 }
 
+// Compute the rate-distortion cost for an encode
+fn compute_tx_rd_cost(
+  fi: &FrameInvariants, fs: &FrameState, w_y: usize, h_y: usize,
+  is_chroma_block: bool, bo: &BlockOffset, bit_cost: u32, tx_dist: i64,
+  bit_depth: usize,
+  skip: bool, luma_only: bool
+) -> f64 {
+  assert!(fi.config.tune == Tune::Psnr);
+  //let lambda = get_tx_lambda(fi, bit_depth);
+  let lambda = get_lambda(fi, bit_depth);
+
+  // Compute distortion
+  let mut distortion = if skip {
+    let po = bo.plane_offset(&fs.input.planes[0].cfg);
+
+    sse_wxh(
+      &fs.input.planes[0].slice(&po),
+      &fs.rec.planes[0].slice(&po),
+      w_y,
+      h_y
+    )
+  } else {
+    assert!(tx_dist >= 0);
+    tx_dist as u64
+  };
+
+  if !luma_only {
+    if skip {
+      let PlaneConfig { xdec, ydec, .. } = fs.input.planes[1].cfg;
+
+      let mask = !(MI_SIZE - 1);
+      let mut w_uv = (w_y >> xdec) & mask;
+      let mut h_uv = (h_y >> ydec) & mask;
+
+      if (w_uv == 0 || h_uv == 0) && is_chroma_block {
+        w_uv = MI_SIZE;
+        h_uv = MI_SIZE;
+      }
+
+      // Add chroma distortion only when it is available
+      if w_uv > 0 && h_uv > 0 {
+        for p in 1..3 {
+          let po = bo.plane_offset(&fs.input.planes[p].cfg);
+
+          distortion += sse_wxh(
+            &fs.input.planes[p].slice(&po),
+            &fs.rec.planes[p].slice(&po),
+            w_uv,
+            h_uv
+          );
+        }
+      }
+    } else {
+
```

Comment:
Nothing to do here?

Reply:
If not luma only and not skipped, then tx_dist is assumed to have the distortion for both luma and chroma.
I have revised to combine "if !luma_only" and "if skip" as one condition.